java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -242,116 +301,164 @@ JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getOutputInfo
+                                                                  jlong sessionHandle, jlong allocatorHandle,
+                                                                  jobjectArray inputNamesArr, jlongArray tensorArr,
+                                                                  jlong numInputs, jobjectArray outputNamesArr,
+                                                                  jlong numOutputs, jlong runOptionsHandle) {
+
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtAllocator* allocator = (OrtAllocator*)allocatorHandle;
+  OrtSession* session = (OrtSession*)sessionHandle;
+  OrtRunOptions* runOptions = (OrtRunOptions*)runOptionsHandle;
+
+  jobjectArray outputArray = NULL;
+
+  // Create the buffers for the Java input & output strings, and the input pointers
+  const char** inputNames = NULL;
+  OrtErrorCode code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(char*) * numInputs, (void**)&inputNames));
+  if (code != ORT_OK) {
+    // Nothing to cleanup, return and throw exception
+    return outputArray;
+  }
+  const char** outputNames = NULL;
+  code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(char*) * numOutputs, (void**)&outputNames));
+  if (code != ORT_OK) {
+    goto cleanup_input_names;
+  }
+  jobject* javaInputStrings = NULL;
+  code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(jobject) * numInputs, (void**)&javaInputStrings));
+  if (code != ORT_OK) {
+    goto cleanup_output_names;
+  }
+  jobject* javaOutputStrings = NULL;
+  code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(jobject) * numOutputs, (void**)&javaOutputStrings));
+  if (code != ORT_OK) {
+    goto cleanup_java_input_strings;
+  }
+  const OrtValue** inputValues = NULL;
+  code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(OrtValue*) * numInputs, (void**)&inputValues));
+  if (code != ORT_OK) {
+    goto cleanup_java_output_strings;
+  }
+  OrtValue** outputValues = NULL;
+  code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(OrtValue*) * numOutputs, (void**)&outputValues));
+  if (code != ORT_OK) {
+    goto cleanup_input_values;
+  }
 
-    // Extract the names of the output values, and allocate their output array.
-    OrtValue** outputValues;
-    checkOrtStatus(jniEnv,api,api->AllocatorAlloc(allocator,sizeof(OrtValue*)*numOutputs,(void**)&outputValues));
-    for (int i = 0; i < numOutputs; i++) {
-        javaOutputStrings[i] = (*jniEnv)->GetObjectArrayElement(jniEnv,outputNamesArr,i);
-        outputNames[i] = (*jniEnv)->GetStringUTFChars(jniEnv,javaOutputStrings[i],NULL);
-        outputValues[i] = NULL;
-    }
+  // Extract a C array of longs which are pointers to the input tensors.
+  // Need to convert longs to OrtValue* in case we run on non-64bit systems
```

Comment:
>Need to convert longs to OrtValue* in case we run on non-64bit systems [](http://example.com/codeflow?start=5&length=70)

I think this stmt requires clarification. I would expect the code to be the same on all arch.

Reply:
In Java we represent all pointers as longs, but on 32-bit architectures the upper 32-bits of those longs will be zero'd and an array of longs is not easily convertible to an array of pointers as the bits are in the wrong place. This logic copies out the array of longs so it's an array of right size pointers.

Reply:
We need to put this wording in the comment. The names still need to be changed tensorArr -> ortValueArr.
The old name inputValues was correct.

Reply:
Ok, I'll update it.